src/routes/did.rs

```diff
@@ -0,0 +1,192 @@
+use actix_session::Session;
+use actix_web::{post, web, HttpResponse};
+use sha2::{Sha512, Digest};
+use sp_runtime::traits::Verify;
+use base64::{Engine, engine::general_purpose};
+use std::sync::RwLock;
+use subxt::ext::codec::{Decode, IoReader};
+
+use crate::{
+    constants::OIDC_SESSION_KEY,
+    kilt::{
+        self, get_did_doc,
+        runtime_types::did::did_details::{DidPublicKey, DidVerificationKey},
+    },
+    routes::error::Error,
+    verify::{hex_decode, hex_encode},
+    AppState, AuthorizeQueryParameters
+};
+
+
+#[derive(serde::Deserialize)]
+#[allow(dead_code)]
+struct JWTHeader {
+    alg: String,
+    typ: String,
+    pub key_uri: String,
+
+}
+
+#[derive(serde::Deserialize)]
+#[allow(dead_code)]
+struct JWTPayload {
+    
+    pub iss: String,
+    pub sub: String,
+    nonce: String,
+    
+}
+
+
+
+#[post("/api/v1/did/{token}")]
+async fn login_with_did(
+    app_state: web::Data<RwLock<AppState>>,
+    session: Session,
+    path: web::Path<String>,
+) -> Result<HttpResponse, Error> {
+    log::info!("POST DID handler");
+
+    let jwt_token = path.into_inner();
+
+    // get requirements for this client, the client id comes from the session
+    let oidc_context = session
+        .get::<AuthorizeQueryParameters>(OIDC_SESSION_KEY)
+        .map_err(|_| Error::OauthNoSession)?
+        .ok_or(Error::OauthInvalidClientId)?;
+
+    let nonce = oidc_context.nonce;
+
+    let mut hasher =  Sha512::new();
+
+    let parts : Vec<&str> = jwt_token.split('.').collect();
```

Comment:
Bette check if you really have three parts, otherwise the [2] could panic

Reply:
Always forget safe array operations :face_exhaling: [e289e12](https://github.com/KILTprotocol/opendid/pull/28/commits/e289e129e41c0e4639110fb340b9ae8736edb8c4)